From d4ff7df4afc97366b3a80d1f79b828f152b0bf4e Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Mon, 5 Apr 2021 13:05:14 +0700
Subject: [PATCH] qtdeclarative: switch to use python3 instead of python2

due to the current filesystem does not support to use python2, we need
to change to python3

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 qtdeclarative.pro          | 2 +-
 src/3rdparty/masm/masm.pri | 4 ++--
 tests/manual/v4/test262.py | 2 +-
 tests/manual/v4/tests.pro  | 4 ++--
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/qtdeclarative.pro b/qtdeclarative.pro
index 5b94da9b69..0e3496ce80 100644
--- a/qtdeclarative.pro
+++ b/qtdeclarative.pro
@@ -2,7 +2,7 @@ CONFIG += tests_need_tools examples_need_tools
 load(qt_parts)
 
 !python_available {
-    py_out = $$system('python -c "print(1)"')
+    py_out = $$system('python3 -c "print(1)"')
     !equals(py_out, 1): error("Building QtQml requires Python.")
     tmp = python_available
     CONFIG += $$tmp
diff --git a/src/3rdparty/masm/masm.pri b/src/3rdparty/masm/masm.pri
index 04548fe8a3..c81537a4ec 100644
--- a/src/3rdparty/masm/masm.pri
+++ b/src/3rdparty/masm/masm.pri
@@ -47,7 +47,7 @@ contains(DEFINES, WTF_USE_UDIS86=1) {
     udis86.output = udis86_itab.h
     udis86.input = ITAB
     udis86.CONFIG += no_link
-    udis86.commands = python $$PWD/disassembler/udis86/itab.py ${QMAKE_FILE_IN}
+    udis86.commands = python3 $$PWD/disassembler/udis86/itab.py ${QMAKE_FILE_IN}
     QMAKE_EXTRA_COMPILERS += udis86
 
     udis86_tab_cfile.target = $$OUT_PWD/udis86_itab.c
@@ -83,7 +83,7 @@ retgen.output = $$GENERATEDDIR/RegExpJitTables.h
 retgen.script = $$PWD/create_regex_tables
 retgen.input = retgen.script
 retgen.CONFIG += no_link
-retgen.commands = python $$retgen.script > ${QMAKE_FILE_OUT}
+retgen.commands = python3 $$retgen.script > ${QMAKE_FILE_OUT}
 QMAKE_EXTRA_COMPILERS += retgen
 
 # Taken from WebKit/Tools/qmake/mkspecs/features/unix/default_post.prf
diff --git a/tests/manual/v4/test262.py b/tests/manual/v4/test262.py
index 3b5bfa119a..f3a0413f10 100755
--- a/tests/manual/v4/test262.py
+++ b/tests/manual/v4/test262.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #############################################################################
 ##
 ## Copyright (C) 2015 The Qt Company Ltd.
diff --git a/tests/manual/v4/tests.pro b/tests/manual/v4/tests.pro
index a86a6bf6af..68399da6aa 100644
--- a/tests/manual/v4/tests.pro
+++ b/tests/manual/v4/tests.pro
@@ -4,12 +4,12 @@ TESTSCRIPT=$$PWD/test262.py
 V4CMD = qmljs
 
 checktarget.target = check
-checktarget.commands = python $$TESTSCRIPT --command=$$V4CMD --parallel --with-test-expectations --update-expectations
+checktarget.commands = python3 $$TESTSCRIPT --command=$$V4CMD --parallel --with-test-expectations --update-expectations
 checktarget.depends = all
 QMAKE_EXTRA_TARGETS += checktarget
 
 checkmothtarget.target = check-interpreter
-checkmothtarget.commands = python $$TESTSCRIPT --command=\"$$V4CMD --interpret\" --parallel --with-test-expectations
+checkmothtarget.commands = python3 $$TESTSCRIPT --command=\"$$V4CMD --interpret\" --parallel --with-test-expectations
 checkmothtarget.depends = all
 QMAKE_EXTRA_TARGETS += checkmothtarget
 
-- 
2.17.1

